api/src/main/java/org/apache/iceberg/ContentFile.java

```diff
@@ -165,6 +166,20 @@ default Long fileSequenceNumber() {
    */
   F copyWithoutStats();
 
+  /**
+   * Copies this file with only specific column stats. Manifest readers can reuse file instances;
+   * use this method to copy data and only copy specific stats when collecting files.
+   *
+   * @param requestedColumnIds column ids for which to keep stats. If <code>null</code> then every
+   *     column stat is kept.
+   * @return a copy of this data file, with stats lower bounds, upper bounds, value counts, null
+   *     value counts, and nan value counts for only specific columns.
+   */
+  default F copyWithStats(Set<Integer> requestedColumnIds) {
```

Comment:
I think it is fine to use IDs at this level as `ContenFile` references IDs everywhere. I agree with converting column names to IDs in scans.

Reply:
Also, if we want to support `null` as a valid value, do we still need `ContentFileUtil`?

```
copyWithStats(null) -> copy with all stats
copyWithStats(empty) -> copy without stats
copyWithStats(someIds) -> copy with some stats
```

It seems we need to either support `null` here or have `ContenFileUtil`, is there a reason to have both? I am not sure what I prefer more, though.

Reply:
I spend a bit more time on this one. I do think we either drop support for `null` as a valid value here or drop the utility to copy. It is up to you which approach to pick.

Reply:
I am not fan of the special value settings, where `null`, and `empty` have specific meaning, so I ended up not defining what we are doing  with `null` on the API.

Still behind the scenes in the constructors for `BaseFile`, `GenericDataFile`, etc, I need to keep the `null` as a possible value, but at least that is not a public API.

Will talk about this one more time, before finalising the solution...